usb/imx-usbc/client.c

```diff
@@ -132,6 +131,22 @@ static void usbclient_irqThread(void *arg)
 			desc_classSetup(&imx_common.dc.setup);
 
 		ctrl_lfIrq();
+
+		/* Initialize endpoints */
+		if (imx_common.dc.op == DC_OP_INIT) {
+			ctrl_initQtd();
+			for (i = 1; i < ENDPOINTS_NUMBER; ++i) {
+				if (ctrl_endptInit(i, &imx_common.data.endpts[i]) < 0) {
+					imx_common.dc.op = DC_OP_INIT_ERR;
+					break;
+				}
+			}
+			if (imx_common.dc.op != DC_OP_INIT_ERR) {
```

Comment:
I think that user should get information that endpoints initialization has failed. What do you think about invoking event with appropriate information?

Reply:
Good point, either/or:
- `cdc_init ()` should wait for this event (with a reasonable timeout e.g. 200ms) and succeed/fail upon receiving a success/failure event after completing the initialization step (not yet descriptor exchange between the host and device), thread worker need to exit, that way `cdc_init()` may call `cdc_destroy()` in itself, and return error.
- `cdc_init ()` works as it is now, but only the appropriate event is sent to the user.

The first approach may be better but requires an another (separate) PR, the second is simpler to implement and may be an amendment to the current PR, but from my personal point of view the first is how to do it (the right way), what do you think?

Reply:
I agree with you. The first approach covers case when the device is not connected to the host and this solution with timeout is recommended.
If you consider that this approach is more complicated you can add the second one with appropriate information in comment that it needs to be changed in the future.

Reply:
Done, according to the first approach.